### ISSUE TITLE
Add flag to build environment.yml without build strings

### ISSUE THREAD

**Issue by kalefranz** [conda/conda#6220]
https://gitter.im/conda/conda?at=59ef54ebe44c43700a70e9a4
https://twitter.com/drvinceknight/status/922837449092542464?ref_src=twsrc%5Etfw

> Due to hashes of packages being introduced in `envinronment.yml` I'm getting all sorts of issues with building envs from file. (Very new problem)

**Comment by kalefranz** [comment]
Actually, looks like the flag might already be there as `--no-builds`

```
kfranz@0283:~/continuum/conda *4.3.x ❯ /conda/bin/conda env export --help
usage: conda-env export [-h] [-c CHANNEL] [--override-channels]
                        [-n ENVIRONMENT | -p PATH] [-f FILE] [--no-builds]
                        [--ignore-channels] [--json] [--debug] [--verbose]

Export a given environment

Options:

optional arguments:
  -h, --help            Show this help message and exit.
  -c CHANNEL, --channel CHANNEL
                        Additional channel to include in the export
  --override-channels   Do not include .condarc channels
  -n ENVIRONMENT, --name ENVIRONMENT
                        Name of environment (in
                        /conda/envs:/Users/kfranz/.conda/envs).
  -p PATH, --prefix PATH
                        Full path to environment prefix (default: /conda).
  -f FILE, --file FILE
  --no-builds           Remove build specification from dependencies
  --ignore-channels     Do not include channel names with package names.
  --json                Report all output as json. Suitable for using conda
                        programmatically.
  --debug               Show debug output.
  --verbose, -v         Use once for info, twice for debug, three times for
                        trace.

examples:
    conda env export
    conda env export --file SOME_FILE

```

**Comment by kalefranz** [comment]
But the `--no-builds` flag appears not to work, based on a spot test.

**Comment by rfdickerson** [comment]
I am also seeing these build strings appear. For instance, now with and installation and export on an environment.yml file, I see:

```
intel-openmp: 2018.0.0-h68bdfb3_7   anaconda
    libgfortran:  3.0.1-h93005f0_2      anaconda
    mkl:          2018.0.0-h5ef208c_6   anaconda
    numpy:        1.13.3-py36h2cdce51_0 anaconda
    scikit-learn: 0.19.1-py36hffbff8c_0 anaconda
    scipy:        0.19.1-py36h3e758e1_3 anaconda
```

This is problematic, since we often do development on macOS, then export the environment.yml to be used to build a Docker image in a Linux environment for deploying our app.

**Comment by kalefranz** [comment]
The PR that fixes this issue, which is #6221, will be released as part of 4.3.31, likely sometime this week or next.
